src/main/java/seedu/address/model/event/Event.java

```diff
@@ -1,11 +1,149 @@
 package seedu.address.model.event;
 
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.member.Member;
+
 /**
  * Represents an Event in the address book.
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
 public class Event {
-    public Event(){
+    // Identity fields
+    private final String name;
+    private final LocalDate date;
+
+    // Data fields
+    private final HashMap<Member, Boolean> participants = new HashMap<>();
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Event(String name, LocalDate date, Set<Member> participants) {
+        requireAllNonNull(name, date);
+        this.name = name;
+        this.date = date;
+        for (Member m : participants) {
+            this.participants.put(m, false);
+        }
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public LocalDate getDate() {
+        return date;
+    }
+
+    /**
+     * Returns an immutable set of participants, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Member> getParticipants() {
+        return Collections.unmodifiableSet(participants.keySet());
+    }
+
+    /**
+     * Removes a set of members from the event.
+     *
+     * @param members to be removed
+     */
+    public void removeParticipants(Set<Member> members) {
+        for (Member m : members) {
+            participants.remove(m);
+        }
+    }
+
+    /**
+     * Check if a member is part of the event.
+     *
+     * @param member to check
+     * @return boolean indicating if member is included
+     */
+    public boolean isParticipant(Member member) {
```

Comment:
Misleading method name. Sounds like you are checking if the guy is a participant (aka a person), rather than if the guy is participating in the event.

Try changing to isParticipatingInEvent?